pkg/network/tracer/tracer_usm_linux_test.go

```diff
@@ -2185,24 +2191,7 @@ func goTLSDetachPID(t *testing.T, pid int) {
 	}, 5*time.Second, 100*time.Millisecond, "process %v is still traced by Go-TLS after detaching", pid)
 }
 
-// goTLSDetacherWrapper meant to run before the given callback and detach USM GoTLS monitoring from the given PID.
-// It is mainly used in the TLS classification tests, as we need to have a clean slate before running the actual test,
-// and since uprobes are not affected by the calls to `Pause` and `Resume` of the eBPF manager, so we detach from the
-// target process before the setup phase.
-func goTLSDetacherWrapper(pid int, callback func(t *testing.T, ctx testContext)) func(t *testing.T, ctx testContext) {
-	return func(t *testing.T, ctx testContext) {
-		goTLSDetachPID(t, pid)
-		if callback != nil {
-			callback(t, ctx)
-		}
-	}
-}
-
-// goTLSAttacherWrapper meant to run before the given callback and attach USM GoTLS monitoring to the given PID.
-// It is mainly used in the TLS classification tests, as we need to have a clean slate before running the actual test,
-// and since uprobes are not affected by the calls to `Pause` and `Resume` of the eBPF manager, so we detach from the
-// target process before the setup phase, we attach to it before the actual test, and detach from it after the test,
-// to ensure the validation process is not affected by the monitoring.
```

Comment:
I am curious about this removal; more specifically why remove the detacher wrapper and not also the attacher wrapper? For context, we added those wrappers as a workaround for the uprobes not being pause-able until now. With this PR fixing it (a big thank you for this by the way!), I would have expected both wrappers to be unneeded with the bypass feature.

My understanding how tests should run with the bypass feature is that we would:
- Modify the config to allow self-hooking
- Manually hook the test binary itself (this has its issues, but that's outside the scope of this discussion)
- Use pause & resume accordingly in the tests; already managed by the `testProtocolClassificationInner` function.
- At the end of the test: revert the config change & detach the binary.

Reply:
Updated, let me know if it looks OK now.